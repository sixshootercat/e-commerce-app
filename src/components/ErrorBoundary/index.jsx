import React, { Component } from "react";
import { ErrorMessage } from "./styles";

class ErrorBoundary extends Component {
  constructor() {
    super();

    this.state = {
      hasError: false,
    };
  }

  static getDerivedStateFromError() {
    return { hasError: true };
  }

  componentDidCatch(error) {
    console.log(error);
  }

  render() {
    if (this.state.hasError) {
      return (
        <ErrorMessage>
          <h1>There`&apos`s been an error</h1>
        </ErrorMessage>
      );
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
