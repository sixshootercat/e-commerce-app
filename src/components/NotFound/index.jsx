import React from "react";
import { Link } from "react-router-dom";
import { Container, Message } from "./styles";
import { CustomButton } from "components";

const message = "404 - Page Not Found";

const PageNotFound = () => {
  return (
    <Container>
      <Message>{message.toLocaleUpperCase()}</Message>
      <Link to="/">
        <CustomButton data-testid="back-home">Back Home</CustomButton>
      </Link>
    </Container>
  );
};
export default PageNotFound;
