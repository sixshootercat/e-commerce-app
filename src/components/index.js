import CustomButton from "./CustomButton";
import CartItem from "./CartItem";
import CheckoutItem from "./CheckoutItem";
import CollectionItem from "./CollectionItem";
import ErrorBoundary from "./ErrorBoundary";
import FormInput from "./FormInput";
import MenuItem from "./MenuItem";
import NotFound from "./NotFound";
import Spinner from "./Spinner";
import StripeButton from "./StripeButton";

export {
  CustomButton,
  CartItem,
  CheckoutItem,
  CollectionItem,
  ErrorBoundary,
  FormInput,
  MenuItem,
  NotFound,
  Spinner,
  StripeButton,
};
