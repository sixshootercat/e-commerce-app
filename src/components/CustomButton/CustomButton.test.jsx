import React from "react";
import { render, cleanup } from "@testing-library/react";
import { CustomButton } from "components";

afterEach(cleanup);

describe("<CustomButton />", () => {
  it("renders correctly without crashing", () => {
    const { asFragment } = render(<CustomButton />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders button children correctly", () => {
    const { getByText } = render(<CustomButton>Whatever</CustomButton>);
    getByText(/Whatever/i);
  });
});
