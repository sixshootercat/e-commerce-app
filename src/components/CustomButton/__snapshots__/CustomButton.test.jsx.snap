// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> renders correctly without crashing 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ ebgsqd"
  />
</DocumentFragment>
`;
